test/e2e/app_change_test.go

```diff
@@ -89,3 +93,86 @@ spec:
 		require.Equal(t, 2, len(resp2.Tables[0].Rows), "Expected to have 2 app-changes")
 	})
 }
+
+func TestAppKindChangeWithMetadataOutput(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+
+	yaml1 := `
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-config
+  annotations:
+    kapp.k14s.io/versioned: ""
+data:
+  redis-config: |
+    maxmemory 3mb
+    maxmemory-policy allkeys-lru
+`
+
+	yaml2 := `
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kapp-secret-1
+  namespace: kapp-namespace-2
+`
+
+	name := "test-app-change"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	firstDeploy, err := os.CreateTemp(os.TempDir(), "output1")
+	assert.NoError(t, err)
+	secondDeploy, err := os.CreateTemp(os.TempDir(), "output2")
+	assert.NoError(t, err)
+
+	defer func() {
+		os.Remove(firstDeploy.Name())
+		os.Remove(secondDeploy.Name())
+	}()
+
+	logger.Section("deploy app", func() {
+		kapp.RunWithOpts([]string{"deploy", "--app-metadata-file", firstDeploy.Name(), "-f", "-", "-a", name}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+	})
+
+	logger.Section("deploy with changes", func() {
+		kapp.RunWithOpts([]string{"deploy", "--app-metadata-file", secondDeploy.Name(), "-f", "-", "-a", name}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml2)})
+	})
+
+	configMapFirstDeploy, err := ioutil.ReadFile(firstDeploy.Name())
```

Comment:
(I think I forgot to add a comment about this one and at line 158)
*Suggestion*: Use `os.ReadFile()` instead